frontend/tests/controllers/ContestUpdateTest.php

```diff
@@ -1928,4 +1928,93 @@ public function testUpdatePenaltyTypeFromAContestWhenProblemVersionIsUpdated(
         }
         $this->assertSame($runs[0]['penalty'], $expectedPenalty);
     }
+
+    /**
+     * A PHPUnit data provider for testing contest recommended flag permissions.
+     *
+     * @return list<array{0: string, 1: bool}>
+     */
+    public function contestRecommendedPermissionsProvider(): array {
+        return [
+            ['support', true],
+            ['quality-reviewer', false],
+            ['course-curator', false],
+            ['mentor', false],
+            ['group-identity-creator', false],
+            ['certificate-generator', false],
+            ['teaching-assistant', false],
+            ['normal', false],
+        ];
+    }
+
+    /**
+     * Test setting recommended flag with different user roles
+     *
+     * @dataProvider contestRecommendedPermissionsProvider
+     */
+    public function testSetRecommendedFlagPermissions(
+        string $userType,
+        bool $shouldHaveAccess
+    ) {
+        // Get a contest
+        $contestData = \OmegaUp\Test\Factories\Contest::createContest();
+
+        // Create user and add to corresponding group
+        ['identity' => $identity] = \OmegaUp\Test\Factories\User::createUser();
+
+        // Add user to the corresponding group based on type
+        if ($userType !== 'normal') {
+            $groupAlias = 'omegaup:' . $userType;
+            $group = \OmegaUp\DAO\Groups::findByAlias($groupAlias);
+            if (is_null($group)) {
+                throw new \OmegaUp\Exceptions\NotFoundException(
+                    'groupNotFound'
+                );
+            }
```

Comment:
Now PHP 8 has integrated the [null-safe operator](https://php.watch/versions/8.0/null-safe-operator), so it's no longer necessary to add this condition:
```suggestion
            if (is_null($group)) {
                throw new \OmegaUp\Exceptions\NotFoundException(
                    'groupNotFound'
                );
            }
```
```suggestion
```
As you can use it like this:
```php
'group_id' => $group?->group_id,
```

Reply:
got it, Thanks 🙇🏼🙇🏼